### ISSUE TITLE
Revamp colour system (no BC)

### ISSUE THREAD

**Issue by C-Lodder** [uikit/uikit#2874]
UIKit currently has 4 colour variations for most components, however I believe the colouring system is rather poor.

It needs to be made much more dymanic (using an array), so that users can simple add a colour to the array, like so:

**variables.scss**
```
$theme-colours: ();
$theme-colours: map-merge((
    "primary"  : #HEX,
    "success"   : #HEX,
    "warning"  : #HEX,
    "danger"   : #HEX
), $theme-colours);
```

**alert.scss**
```
@each $colour in $theme-colours {
  $key: nth($colour, 1);
  $value: nth($colour, 2);
  
  .alert-#{$key} {
    background-color: $value;
  }
}
```

It makes life much easier when using colour variants for components and simplifies the SCSS.

Same can also be done for LESS, althought I have no idea why LESS is still being supported.

**Comment by mechanic22** [comment]
I agree with this too. It would be awesome to see a way to define our own colour names, instead of using primary & secondary. Be able to use like uk-button-red, uk-button-blue etc. Which is user defined.
